modules/apps/object/object-rest-test/src/testFunctional/macros/OAuth2ForObject.macro

```diff
@@ -0,0 +1,55 @@
+definition {
+
+	macro addAuth2Application {
+		if (isSet(baseURL)) {
+			SignIn.signInSpecificURL(url = ${baseURL});
+		}
+		else {
+			var baseURL = PropsUtil.get("portal.url");
+		}
+
+		OAuth2.openOAuth2Admin(baseURL = ${baseURL});
+
+		OAuth2.addApplication(
+			applicationName = "OAuth Application",
+			checkboxUncheckList = "Authorization Code");
+
+		OAuth2.inputApplicationValues(
+			clientId = "myApplication",
+			clientSecret = "mySecret");
+	}
+
+	macro createTokenWithOAuth2Scopes {
+		Variables.assertDefined(parameterList = ${resourceCheckList});
+
+		if (!(isSet(baseURL))) {
+			var baseURL = PropsUtil.get("portal.url");
+		}
+
+		OAuth2.openOAuth2Admin(baseURL = ${baseURL});
+
+		OAuth2.editScopes(
+			applicationName = "OAuth Application",
+			resourceCheckList = ${resourceCheckList},
+			resourcePanels = ${resourcePanels});
+
+		var accessToken = OAuth2.getAccessToken(
+			client_id = "myApplication",
+			client_secret = "mySecret",
+			grantFlow = "clientCredentials",
+			portalURL = ${baseURL});
+
+		return ${accessToken};
+	}
+
+	macro setUpGlobalTokenForDefaultInstance {
```

Comment:
could we divide it here in two methods? first one to create the token and then another one to make it static? The setUpGlobalTokenForDefaultInstance could be called staticDefaultInstanceToken and have the defaultInstanceToken as a parameter

Reply:
I've removed the setUpGlobalTokenForDefaultInstance() which is not called anymore. There is a method createTokenWithOAuth2Scopes to create token which called in testcase file.